mlibrary/pride
# Install `mocha`, `chai`, and `jsdom` for testing via Node.

# Overview
One of the end goals of rebuilding `pride` is to use GitHub Actions to increase QA of each pull request, which will include running unit tests.

This pull request installs `mocha`, `chai`, and `jsdom`, and converts all `*.spec.js` files to run in node. The browser test has been deleted, along with the need for the `./spec/pride.spec.js` build file. Scripts have been rewritten and the new `test` script has been added.

## Anything else?
Documentation has been updated to provide clearer testing practices.

## Testing
1. Build the repository (`npm run build`).
2. Run `npm run test` to see if tests pass.
   * Break the new/updated unit tests to make sure they are working properly.
3. Navigate to your local [Search repository](https://github.com/mlibrary/search) and apply the newly generated files:
   1. Open `./package.json` and add `#branch-name-of-pull-request` at the end of the `pride` dependency URL:
      ```bash
      "pride": "git+https://github.com/mlibrary/pride.git#install-mocha-chai"
      ``` 
   2. Do a clean install of Search:
      ```bash
      rm -rf node_modules && rm package-lock.json && npm install
      ``` 
   3. Start Search (`npm start`) and look around [the site](http://localhost:3000/everything). Check to see if there are any console errors, and everything still works as expected.
- _How to use the feature_.
- Make sure the PR is consistent in these browsers:
